Golem/Golem.cs

```diff
@@ -451,5 +479,88 @@ public List<String> LogFiles()
         {
             return Provider.ProviderProcess?.Id;
         }
+
+        private async Task SetupSuspendHandlerAsync()
+        {
+            await SuspendHandlerLock.WaitAsync();
+            try {
+                if (_isSuspendHandlerSetUp) {
+                    return;
+                }
+    #pragma warning disable CA1416 // Validate platform compatibility ensured by #if
+                SetupSuspendHandler_Impl();
+    #pragma warning restore CA1416 // Validate platform compatibility
+                _isSuspendHandlerSetUp = true;
+            } finally {
+                SuspendHandlerLock.Release();
+            }
+
+        }
+        private async Task DetachSuspendHandler()
+        {
+            await SuspendHandlerLock.WaitAsync();
+            try {
+#pragma warning disable CA1416 // Validate platform compatibility ensured by #if
+            DetachSuspendHandler_Impl();
+#pragma warning restore CA1416 // Validate platform compatibility
+            _isSuspendHandlerSetUp = false;
+            } finally {
+                SuspendHandlerLock.Release();
+            }
+        }
+
+        #if OS_WINDOWS
```

Comment:
See https://stackoverflow.com/a/975370 - it claims that
"""
If you use conditional compilation on the other hand you can still leave all pieces of code that call the method. All parameters will still be validated by the compiler. The method just won't be called at runtime. I think that it is way better to hide the method just once and not have to remove all the code that calls it as well. You are not allowed to use the conditional attribute on methods which return value - only on void methods. But I don't think this is a big limitation because if you use #if with a method that returns a value you have to hide all pieces of code that call it too.
"""
So it looks like disabling warnings isn't needed then.

Reply:
Let's allow it to avoid having to retest it as it's a time-intensive process.